simpleclient_spring_boot/src/test/java/io/prometheus/client/spring/EnableCollectorRegistrationAnnotationTests.java

```diff
@@ -0,0 +1,129 @@
+import io.prometheus.client.Collector.MetricFamilySamples.Sample;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+import io.prometheus.client.spring.CollectorRegistrationBeanPostProcessor;
+import io.prometheus.client.spring.CollectorRegistryAware;
+import io.prometheus.client.spring.CollectorRegistryAwareBeanPostProcessor;
+import io.prometheus.client.spring.CollectorRegistryFactoryBean;
+import io.prometheus.client.spring.EnablePrometheusCollectorRegistration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Stuart Williams (pidster)
+ */
+@RunWith(SpringRunner.class)
+public class EnableCollectorRegistrationAnnotationTests {
+
+    @Autowired
+    ApplicationContext context;
+
+    @Test
+    public void testEnableAnnotation() {
+        assertThat(context.getBeansOfType(CollectorRegistryFactoryBean.class)).hasSize(1);
+        assertThat(context.getBeansOfType(CollectorRegistryAwareBeanPostProcessor.class)).hasSize(1);
+        assertThat(context.getBeansOfType(CollectorRegistrationBeanPostProcessor.class)).hasSize(1);
+        assertThat(context.getBeansOfType(TestCollectorRegistryAware.class)).hasSize(1);
+
+        TestCollectorRegistryAware aware = context.getBean(TestCollectorRegistryAware.class);
+        assertNotNull(aware.collectorRegistry);
+
+        CollectorRegistry registry = context.getBean(CollectorRegistry.class);
+        assertNotNull(registry);
+        assertNotEquals(CollectorRegistry.defaultRegistry, registry);
+
+        List<Sample> sampleList = new ArrayList<Sample>();
+
+        Enumeration<MetricFamilySamples> samplesEnum = registry.metricFamilySamples();
+        while (samplesEnum.hasMoreElements()) {
+            MetricFamilySamples element = samplesEnum.nextElement();
+            sampleList.addAll(element.samples);
+        }
+
+        assertThat(sampleList).hasSize(1);
+        assertEquals("testnamespace_testsubsystem_testname", sampleList.get(0).name);
+    }
+
+    @Test
+    public void testEnableAnnotationClasses() {
+        assertThat(context.getBeansOfType(CollectorRegistryFactoryBean.class)).hasSize(1);
+        assertThat(context.getBeansOfType(CollectorRegistry.class)).hasSize(1);
+        assertThat(context.getBeansOfType(CollectorRegistryAwareBeanPostProcessor.class)).hasSize(1);
+        assertThat(context.getBeansOfType(CollectorRegistrationBeanPostProcessor.class)).hasSize(1);
+    }
+
+    @Test
+    public void testCollectorRegistryAware() {
+        TestCollectorRegistryAware aware = context.getBean(TestCollectorRegistryAware.class);
+        assertNotNull(aware.collectorRegistry);
+
+        CollectorRegistry registry = context.getBean(CollectorRegistry.class);
+        assertNotNull(registry);
+        assertNotEquals(CollectorRegistry.defaultRegistry, registry);
```

Comment:
What is the `CollectorRegistryAware` needed for actually?  As we show here the `CollectorRegistry` can be received either from `CollectorRegistryAware` or from `context`. In case of `context`, I assume it should be possible to use simple `@Autowired CollectorRegistry collectorRegistry`. 
I'm wondering if the `CollectorRegistryAware`  is actually required in some cases or if this is just an alternative to `@Autowired`.